src/services/pure-di.tsx

```diff
@@ -176,6 +176,7 @@ export class AppCompositionRoot implements AppServices {
           appSessionRef: this.appSessionRef as MutableRefObject<AppSession>,
           dopplerLegacyClient: this.dopplerLegacyClient,
           keepAliveMilliseconds: this.appConfiguration.dopplerLegacyKeepAliveMilliseconds,
+          maintenanceAutoDetect: process.env.REACT_APP_ALLOW_MANTENAINCE_AUTODETECT === 'true',
```

Comment:
I am not totally sure, but it seems that the old approach is using `this.appConfiguration` in place of directly reading `process`. 

I think that it could be useful if we want to override the complete configuration. I mean, having a single point for reading the configuration.

![image](https://user-images.githubusercontent.com/1157864/100223809-bea33200-2efa-11eb-9eb9-f18563f28c8c.png)


Another advantage is that you unify the parsing and typing in a single place:

```suggestion
          maintenanceAutoDetect: this.appConfiguration.allowMantenainceAutodetect,
```

```javascript
  get appConfiguration() {
    return this.singleton('appConfiguration', () => ({
      allowMantenainceAutodetect: process.env.REACT_APP_ALLOW_MANTENAINCE_AUTODETECT === 'true',
      dopplerLegacyUrl: process.env.REACT_APP_DOPPLER_LEGACY_URL as string,
      dopplerSitesUrl: process.env.REACT_APP_DOPPLER_SITES_URL as string,
```

Reply:
We have all kinds of use of `process.env.....` everywhere, I will fix this here and be on the lookout for other mishaps.

Reply:
> We have all kinds of use of `process.env...`

We should not. It should be reduced to the minimum.